### ISSUE TITLE
Network issues when enabled, when switching wifi/cellular/wifi and randomly on new Samsung s24 ultra

### ISSUE THREAD

**Issue by cocokola** [AdguardTeam/AdguardForAndroid#5429]
### Please answer the following questions for yourself before submitting an issue

- [X] Filters were updated before reproducing an issue
- [X] I checked the [knowledge base](https://adguard.com/kb/) and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

4.3.1

### Environment

- OS version: android 14, ui 6.1,
![SmartSelect_20240810_122034_Settings](https://github.com/user-attachments/assets/89eaf4d3-d94e-4daa-ad02-31483fdc010f)

- Device:
Samsung s24 ultra

### HTTPS filtering

- [X] yes, I do

### Root access

- [ ] yes, I have it

### Integration with AdGuard VPN

- [ ] yes, I do

### Routing mode

Local VPN

### Ad Blocking

AdGuard Base filter, AdGuard Mobile Ads filter

### Privacy

AdGuard Tracking Protection filter

### Social

AdGuard Social Media filter

### Annoyances

AdGuard Annoyances filter

### Other

AdGuard DNS filter

### Which DNS server do you use?

Automatic DNS

### DNS protocol

None

### Issue Details

Steps to reproduce:
Open browser
Enable adguard .  Switch from cellular to wifi.  Try to surf to www.google.com,  app will time out.  Disable adguard, works as expected.

### Expected Behavior

On my note10 plus,  browsing worked as excited no matter the connection.  I do recall going to expert features  months ago and bypassing rules for local subnet, which fixed some oddities, but morning else.

### Actual Behavior

Apps time out, browser times out

### Screenshots

<details><summary>Screenshot 1</summary>

</details>

### Additional Information

Not sure what to try.  I installed app, added beta key and just have issues.   It's got to be me, but i can't figure out what.

Your questions about what is enabled does not fit with the new gui.

**Comment by maxikuzmin** [comment]
@cocokola please clarify, when surfing to google.com - which app time out? Browser or AdGuard? Which browser are you using? Does this problem recur with other browsers?

**Comment by cocokola** [comment]
A plethora of apps.  Random issues.    I am beginning to recall, on the older version, there was a way to watch each connection, see what was blocked or allowed and for which apps.  Whenever I had an issue where turning off Adguard fixed it, I would go into this stream of data and find things related to that app, add rules until it worked, and then remove rules until I broke it again to minimize plugs added to the firewall.   On this version, I did find a similar feature, but it only allowed me to add a block ( or was it remove a block, i don't recall other than it was the opposite of what I needed to open up the rules to allow the app to work.   My only workaround is bypassing the entire app, which is not ideal.

browsing works, I am talking about the 1700+ apps I have on my phone.  It took several weeks of fiddling with rules until it was working.   On this version, the majority of apps have issues.

One example: is Google Music:   I can select an artist, go to their top songs, click on play all, or play a particular song. which it does, but then stops, and gets an error message I am offline.  I can press the next song manually and it will play, but only that one song.  bypassing adguard, or disabling it fixes this issue.  

make sense?  sorry I had covid recently, it scrambled my brain making communicating difficult, so please be patient with me. thanks!

**Comment by cocokola** [comment]
I just switched from my 5gh wifi to 2.4,  and back, google news feed stopped working for 4 minutes,now it's working again.

And why is adguard blocking my watch from dns lookup?

And why can't I allow it from this view like before?


![Screenshot_20240812_234041_AdGuard](https://github.com/user-attachments/assets/cb34ac65-5467-4139-9025-43068dd74a8a)
![Screenshot_20240812_233952_Chrome](https://github.com/user-attachments/assets/4c829e48-3227-4da7-9675-25d430df0fc6)

**Comment by cocokola** [comment]
If I wait long enough,somr connections starts
 Working.  

Well, i haven't confirmed this is the cause, but it's very likely.   Both the beta version from app store and the version download from adguard were installed.

It's confusing there is a beta version, but should I not be using the app from adguard for full features?

Thank you and my perhaps prematue apologies.  I think when I used Samsung switch it loaded the appstore version. I dunno

**Comment by Schism86** [comment]
I have the same problem with my Poco F3 with Custom Rom crDroid 10.7 from August 12, 2024. 
What I noticed is that the internet connection is working again after I did a filter update.

**Comment by KobeLAL** [comment]
I have the same problem, but only when I switch from the wifi to the mobile network. Then I can't open any pages in the browser and no app has an internet connection. If I deactivate Adguard, I can access the Internet normally again. Pixel 6 Android 15.

**Comment by cocokola** [comment]
Still my questions about ability to edit rules in effect when a block stops an app from working

**Comment by Eugene-Savenko** [comment]
A similar complaint from support (CRM ID 932631):

Today, I came home after using the internet outside, and once again, neither the browser, VK, nor any other app had internet access, even though WiFi was working and connected. This issue started right after I was advised to enable the "Always connect to VPN" feature, following my complaint that AdGuard was not starting automatically after the phone reboot on an earlier version.

At around 11:20 AM, I left the house, and at 12:17 PM, the phone connected to the home WiFi, but neither the browser nor online apps had an active connection. Chrome repeatedly tried to connect to the Google homepage. I tried to pause AdGuard from the notification panel, but instead opened the app, and the connection immediately appeared.

**Comment by cocokola** [comment]
Perhaps. 

I have system-wide issues all the time.  For example printing, I've disabled adguard for every system service and app involved, bypassing adguard l, but I'm still unable to print normally. 
If adguard is enabled, on system, disabled on all print services, i get these errors: 
![Screenshot_20240820_190251_One UI Home](https://github.com/user-attachments/assets/c7056ffd-ed3e-486a-9aa2-d4554b249d39)
![Screenshot_20240820_190244_One UI Home](https://github.com/user-attachments/assets/05ef20a6-8441-4de8-b839-66d05cac2003)

To be clear, i didn't intend on closing these bugs.

Should i create bug report for every app or process with an issue?  
I need some help to make this app usable.  I am constantly turning it on off to get through each day

**Comment by cocokola** [comment]
Another example
![Screenshot_20240820_194303_Google](https://github.com/user-attachments/assets/2161fde9-4634-47c2-a141-396c213551f7)

**Comment by cocokola** [comment]
![20240820_194516](https://github.com/user-attachments/assets/88ad0ace-1598-4098-83fc-d977713a4cfb)
Another example. 
Is the working floor anyone else with s24 ultra having these issues?

**Comment by Eugene-Savenko** [comment]
@cocokola How is your printing issue related to the one being discussed here? Please use spoilers for images.

**Comment by FutureSharks** [comment]
I had the same problem, it was consistently reproducible for me but then the issue went away. I'll send logs when I can see the issue.

**Comment by KobeLAL** [comment]
@maxikuzmin: I am on the current nightly build. I have activated logging and will try to reproduce the error.

**Comment by cocokola** [comment]
Received An update notification  this morning, updated. I'll try to
reproduce further.

I disabled tracking and annoinces and some things improved, but I'll start
over.

Happy to help squash bugs.
1) is there a way to bypass or unblock individual blocked items?
2)  if you disabled this ability to flush out problems let me know to
report everything i run into.

Thanks!

**Comment by maxikuzmin** [comment]
@cocokola to unlock individual items, you can create User rules, [here](https://adguard.info/kb/general/ad-filtering/create-own-filters/#basic-rules) is all the info on that. Also you can find the blocked item in the filtering log, tap on it, and tap the “Add allowing rule” button. Or you can add the website in allowlist.

@FutureSharks @KobeLAL @cocokola were you able to reproduce the problem again and record logs?

**Comment by FutureSharks** [comment]
> were you able to reproduce the problem again and record logs?

Sadly not. When I get the problem, once I go into settings it stops happening.

**Comment by starbase64** [comment]
Hi, i have same problems on Samsung S24 Ultra. No internet connection over wifi if adguard is enabled, mobile connection works properly.

**Comment by maxikuzmin** [comment]
@starbase64 please record the logs and email them to us as per the instructions above

**Comment by cocokola** [comment]
I'll get these to you.   I have a disability hearing in a month, I'm busy
getting it all together. Thanks for your patience.

In the interim, can my earlier questions be answered?

**Comment by FutureSharks** [comment]
I had the issue this morning with the debug logs enabled also. I've emailed the zip file.

**Comment by cocokola** [comment]
I had printing issues, logs sent.

I missed that you responded how to adjust rules  I'll try that now as well .


I do know turning off private tracking and annoyances did help fix some
issues.  I had so many it was difficult to track

**Comment by maxikuzmin** [comment]
@cocokola @KobeLAL @FutureSharks in the upcoming 4.6.3 update we will fix this bug. Thanks for your logs. 
We previously had this problem and fixed #4694 #4797 , but unfortunately it has reappeared again

**Comment by cocokola** [comment]
Pleased to help.  I'll have many more to share.  Must i open a new ticket
for each app issue?  I understand need to close out work to prioritize,  do
these reopen with new contentn posted, such as this, or shall i open new,
refer to this ticket for historical?
Thank you for being so responsive.

**Comment by cocokola** [comment]
Why are some blocked connections unable to be allowed? See screen shots.

 filtering log =recent activity.  That's not clear/labeled.

It's strange out is under statistics.  I would expect new users to look for
a main icon in the bottom that clearly states activity log to make it more
new user friendly, .. my two cents..

**Comment by maxikuzmin** [comment]
@cocokola @KobeLAL @FutureSharks we have fixed this issue in the new version 4.6.3. Please update and check again for network issue

**Comment by maxikuzmin** [comment]
@cocokola yes please for every problem you need to open a new issue. If you also want to suggest something for improvement, please open a new Feature request and we'll look into it
